### ISSUE TITLE
Merge with the efforts on ZF-Commons / zfc-rbac 

### ISSUE THREAD

**Issue by FrankHouweling** [Laminas-Commons/LmcRbac#11]
We are currently having an option discussion on the future of zfc-rbac on the relevant github project issues page, and I would very much appreciate your opionion on the matter. Some individuals, myself included, would very much prefer to merge the two projects together and move towards a shared goal.

The relevant issue:
https://github.com/ZF-Commons/zfc-rbac/pull/392

**Comment by basz** [comment]
In that discussion, we sort of agreed on the following https://github.com/ZF-Commons/zfc-rbac/pull/392#issuecomment-650177476

This organization would need two repositories instead of one to hold the original zfc-rbac and it v3 continuation.

Of course we would also need commit access... Thanks!

**Comment by visto9259** [comment]
When I looked at the different branches, I also though that maybe it would be good idea to separate the Rbac component from the MVC components (guards, strategies, etc.).

I am touching base with @matwright  on getting more people on board.  I do not think this is an issue as we did not intend to "own" Laminas-Commons.

**Comment by visto9259** [comment]
[Laminas-Commons/LmcRbacMvc](https://github.com/Laminas-Commons/LmcRbacMvc) has been created to hold the former ZfcRbac v2.x package.
This repository will hold the former ZfcRbac v3.

**Comment by svycka** [comment]
maybe someone could write what changed what migrated how names or repositories changed... hard to track everything in all the issues may be merged v3 branch into master if this is 3.0 version? and release 3.0 I think it is pretty stable just documentation is missing

**Comment by visto9259** [comment]
The efforit to bring a migration to Laminas of the former ZfcRbac v3 towards Laminas has not been done yet .  So yes, it is confusing at this time since the master branch still holds the equivalent of ZfcRbac v2.

So what needs to happen, is to bring a Laminas migration of ZfcRbac v3 into this repo, remove any v2 branches and tags (or keep them for history?) and have the master branch point to ZfcRbac v3.

I lack the bandwith to do this right now.  If there are volunteers, I can add them as members of this repo's dev team.

**Comment by basz** [comment]
I can have a go at it. Work is mostly done in a PR somewhere...

**Comment by visto9259** [comment]
@basz, added you to the @Laminas-Commons/lmcrbac-dev team
Let me know that you got the invitation.

**Comment by basz** [comment]
Yep, got it. Thank you... Stupid question maybe - but just to be clear - this repo should become https://github.com/ZF-Commons/zfc-rbac/tree/develop... correct?

Cause then I would expect that branch to be present in this repo under https://github.com/Laminas-Commons/zfc-rbac/tree/develop or https://github.com/Laminas-Commons/zfc-rbac/tree/laminas-3.x but those seem variations on https://github.com/Laminas-Commons/zfc-rbac/tree/master... version 2.

**Comment by svycka** [comment]
I think it's too much confusion with versions maybe just remove all tags and make it version 1.0? as this is an almost complete rewrite and we had a 3.0 beta version and now we have 2.0 version which will be 3.0 after laminas migration? so beta or alpha version can't remember becomes 4.0 from 3.0? sorry to many versions... so maybe version 1.0 would be best no?

**Comment by basz** [comment]
Once we decided on a path, first thing I'll do is update the README as it is unclear indeed atm...

**Comment by visto9259** [comment]
@basz, It looks like I may have dropped the original `develop` branch of ZfcRbac 3.  My focus was on porting ZfcRbac v2 until it was decided that this repo was to become the Laminas port of ZfcRbac 3.  I left the work as it was at that point.

It may be best to restart this repo from a fresh import of ZfcRbac.

**Comment by basz** [comment]
I though so. That's fine. 

Think I need to do the following;
- delete master so I can replace it with the develop branch of zfccommon-rbac, that way history of it will be retained. Since I don't have the rights to do that can you do that?
- create a PR that's about ready to merge into the new master (laminas)
- then I'll update namespaces and naming, what you did for the v2
- anything else come after that.

sound ok?

**Comment by visto9259** [comment]
Done.
- `LmcRbac/master` was renamed to `LmcRbac/master_2x`
- `LmacRbac/develop` was renamed to `LmacRbac/develop_2.x
- `LmcRbac/master` should be the same as `ZfcRabc/master`.
- `LmcRbac/develop` should be the same as `ZfcRbac/develop`.

**Comment by visto9259** [comment]
Updated.
- `LmcRbac/master` was renamed to `LmcRbac/master_2x`
- `LmacRbac/develop` was renamed to `LmacRbac/develop_2.x`
- `LmcRbac/master` should be the same as `ZfcRabc/develop`
- `LmcRabc/ZfcRabc/master` is the same as `ZfcRbac/master`
- `LmcRbac/develop` was deleted.

Once the migration work is done and `LmacRbac/master` is released, we can probably delete the other branches.  I just don't like to delete stuff until nobody needs any history.

**Comment by svycka** [comment]
> LmacRbac/develop was renamed to LmacRbac/develop_2.x

I think this is not quite right `LmacRbac/develop` is 3.x and 2.x does not have `develop` branch and anyway, you renamed develop to master in the end :D

**Comment by visto9259** [comment]
> LmacRbac/develop was renamed to LmacRbac/develop_2.x

> I think this is not quite right LmacRbac/develop is 3.x and 2.x does not have develop branch and anyway, you renamed develop to master in the end :D

I was trying to explain the steps I took to move from the original `LmcRbac` repo structure (which was setup more as a repo for ZfcRbac v2) to the current structure.  I did it so that, if anybody was already looking at the repo at that time, they would know what happened to it.

**Comment by basz** [comment]
hi! [1.0.0](https://github.com/Laminas-Commons/LmcRbac/releases/tag/1.0.0) has been released
